probe-rs/src/debug/debug_info.rs

```diff
@@ -893,28 +894,34 @@ impl DebugInfo {
     ///
     /// The unwind loop will continue until we meet one of the following conditions:
     /// - We can no longer unwind a valid PC value to be used for the next frame.
-    /// - We encounter a LR register value of 0xFFFFFFFF which is the 'Reset` value for that register.
+    /// - We encounter a LR register value of 0x0 or 0xFFFFFFFF(Arm 'Reset' value for that register).
```

Comment:
FWIW - there is no defined reset value on ARM.  Each core implementor is allowed to change this value.

In practice what you have here covers 99% of cases I've seen, but I have seen cores where the start is address 0, so the unwind as written would miss the very bottom of frame.

There's not much that can be generally done about this so I don't think it's worth blocking on.  GDB "solves" it by detecting the frame falling out of a valid location or cycling, which is just as confusing.

Reply:
Cool. ARM M7 Technical Reference, A 2.3.1 refers to that as the reset value. They do however have other LR values that they reserve for exception handling that I have not implemented yet. I imagine we might have to also think about it a more granular level than just 'arm' when we add more functionality here.

PS. I definitely think it will be worth to catch the additional frame at some point in the near future.

Reply:
To make it more complicated there are some things that ARM implementors are allowed to change in their "instance" of a standard core.  Initial entry point comes up often in Cortex-A, though I see it less in Cortex-M.

As I said, what you have here covers the 99% case of what I've seen so I wouldn't put much more energy into this right now but if we get bug reports on weird unwinds at the bottom of the stack this is something to keep in mind.

No matter what this has to be heuristic based given the diversity of behavior on market.